ush/gaussian_sfcanl.sh

```diff
@@ -133,63 +132,78 @@ export PGM=$GAUSFCANLEXE
-${NLN} "${FIXorog}/${CASE}/${CASE}.mx${OCNRES}_oro_data.tile4.nc" "./orog.tile4.nc"
-${NLN} "${FIXorog}/${CASE}/${CASE}.mx${OCNRES}_oro_data.tile5.nc" "./orog.tile5.nc"
-${NLN} "${FIXorog}/${CASE}/${CASE}.mx${OCNRES}_oro_data.tile6.nc" "./orog.tile6.nc"
+  # input analysis tiles (with nst records)
+  ${NLN} "${COMIN_ATMOS_RESTART}/${gcycle_date:0:8}.${gcycle_date:8:2}0000.sfcanl_data.tile1.nc" "./anal.tile1.nc"
+  ${NLN} "${COMIN_ATMOS_RESTART}/${gcycle_date:0:8}.${gcycle_date:8:2}0000.sfcanl_data.tile2.nc" "./anal.tile2.nc"
+  ${NLN} "${COMIN_ATMOS_RESTART}/${gcycle_date:0:8}.${gcycle_date:8:2}0000.sfcanl_data.tile3.nc" "./anal.tile3.nc"
+  ${NLN} "${COMIN_ATMOS_RESTART}/${gcycle_date:0:8}.${gcycle_date:8:2}0000.sfcanl_data.tile4.nc" "./anal.tile4.nc"
+  ${NLN} "${COMIN_ATMOS_RESTART}/${gcycle_date:0:8}.${gcycle_date:8:2}0000.sfcanl_data.tile5.nc" "./anal.tile5.nc"
+  ${NLN} "${COMIN_ATMOS_RESTART}/${gcycle_date:0:8}.${gcycle_date:8:2}0000.sfcanl_data.tile6.nc" "./anal.tile6.nc"
 
-${NLN} "${SIGLEVEL}" "./vcoord.txt"
+  # input orography tiles
+  ${NLN} "${FIXorog}/${CASE}/${CASE}.mx${OCNRES}_oro_data.tile1.nc" "./orog.tile1.nc"
+  ${NLN} "${FIXorog}/${CASE}/${CASE}.mx${OCNRES}_oro_data.tile2.nc" "./orog.tile2.nc"
+  ${NLN} "${FIXorog}/${CASE}/${CASE}.mx${OCNRES}_oro_data.tile3.nc" "./orog.tile3.nc"
+  ${NLN} "${FIXorog}/${CASE}/${CASE}.mx${OCNRES}_oro_data.tile4.nc" "./orog.tile4.nc"
+  ${NLN} "${FIXorog}/${CASE}/${CASE}.mx${OCNRES}_oro_data.tile5.nc" "./orog.tile5.nc"
+  ${NLN} "${FIXorog}/${CASE}/${CASE}.mx${OCNRES}_oro_data.tile6.nc" "./orog.tile6.nc"
 
-# output gaussian global surface analysis files
-${NLN} "${COMOUT_ATMOS_ANALYSIS}/${APREFIX}sfcanl.nc" "./sfc.gaussian.analysis.file"
+  ${NLN} "${SIGLEVEL}" "./vcoord.txt"
 
-# Namelist uses booleans now
-if [[ ${DONST} == "YES" ]]; then do_nst='.true.'; else do_nst='.false.'; fi
+  # output gaussian global surface analysis files
+  ${NLN} "${COMOUT_ATMOS_ANALYSIS}/${RUN}.t${gcycle_date:8:2}z.sfcanl.nc" "./sfc.gaussian.analysis.file"
 
-# Executable namelist
-cat <<EOF > fort.41
+  # Namelist uses booleans now
+  if [[ ${DONST} == "YES" ]]; then do_nst='.true.'; else do_nst='.false.'; fi
+
+  # Executable namelist
+  cat <<EOF > fort.41
  &setup
-  yy=${iy},
-  mm=${im},
-  dd=${id},
-  hh=${ih},
+  yy=${gcycle_date:0:4},
+  mm=${gcycle_date:4:2},
+  dd=${gcycle_date:6:2},
+  hh=${gcycle_date:8:2},
   igaus=${LONB_SFC},
   jgaus=${LATB_SFC},
   donst=${do_nst},
  /
 EOF
 
-${APRUNSFC} "${GAUSFCANLEXE}"
+  ${APRUNSFC} "${GAUSFCANLEXE}"
 
-export err=$?
-if [[ ${err} -ne 0 ]]; then
-   echo "FATAL ERROR: ${GAUSFCANLEXE} returned non-zero exit status!"
-   exit "${err}"
-fi
+  export err=$?
+  if [[ ${err} -ne 0 ]]; then
+    echo "FATAL ERROR: ${GAUSFCANLEXE} returned non-zero exit status processing ${gcycle_date}!"
+    # Since we are in "${DATA}/gausfcanl_${gcycle_date:8:2}", we need to step out
+    cd "${DATA}" || exit "${err}"
```

Comment:
```suggestion
    exit "${err}"
```
This is incorrect, as `cd ${DATA}` will be true, and the `||` will not be executed.